src/shared-components/alert/index.tsx

```diff
@@ -87,19 +87,41 @@ export const Alert = (alertProps: AlertProps) => {
   };
 
   useEffect(() => {
-    // Truncate text logic
```

Comment:
I moved the two concerns (truncating text and alert duration) into separate effects for clarity. 

You can reproduce an issue where we get permanently stuck in the `while` loop at https://radiance-ui.curology.com/?path=/story/components-alert--with-image if you toggle between the themes once or twice. You'll notice the text is not truncated. That's because `contentElement.scrollHeight > contentElement.offsetHeight` is always true. This fixes the issue, which Chromatic actually caught in [Build 660](https://www.chromatic.com/build?appId=5f6e46f1e7081000221523cf&number=660).